File: 2017 COMLAN/Logger/src-weaved/matlab/logger_test.m
function logger_test()
   log_file_0 = fopen('log.txt', 'at+');
   fprintf(log_file_0, 'Logging to a file\n');
   fprintf('Print double %f after foo\n', 2.0);
   foo();
   fprintf('Printing again\n');
   fprintf(log_file_0, 'Logging again to a file\n');
   fprintf('Print double %f after bar\n', 2.0);
   fclose(log_file_0);
end

function [a] = foo()
   log_file_1 = fopen('log.txt', 'at+');
   a = 0;
   for i = 0:5
      fprintf(log_file_1, 'Logging to a file\n');
      fprintf('Print double %f after foo\n', 2.0);
      a = a + bar();
      fprintf('Printing again\n');
      fprintf(log_file_1, 'Logging again to a file\n');
      fprintf('Print double %f after bar\n', 2.0);
   end
   fclose(log_file_1);
end

function [result] = bar()
   result = 1.0;
end

File: 2017 COMLAN/Logger/src-weaved/cpp/logger_test.cpp
#include <iostream>
#include <fstream>

double bar() {
   
   return 1.0;
}


double foo() {
   std::ofstream log_file_0;
   log_file_0.open("log.txt", std::ios_base::app);
   double a = 0;
   for(int i = 0; i < 1000; i++) {
      log_file_0 << "Logging to a file" << "\n";
      std::cout << "Print double " << 2 << " after bar" << "\n";
      a += bar();
      std::cout << "Printing again";
      log_file_0 << "Logging again to a file" << "\n";
      std::cout << "Print double " << 2 << " after bar" << "\n";
   }
   
   return a;
}


int main() {
   std::ofstream log_file_1;
   log_file_1.open("log.txt", std::ios_base::app);
   log_file_1 << "Logging to a file" << "\n";
   std::cout << "Print double " << 2 << " after bar" << "\n";
   foo();
   std::cout << "Printing again";
   log_file_1 << "Logging again to a file" << "\n";
   std::cout << "Print double " << 2 << " after foo" << "\n";
}
